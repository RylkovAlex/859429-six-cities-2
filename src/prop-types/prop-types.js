import PropTypes from 'prop-types';

export const offerCardPropTypes = {
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitide: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
  }),
  id: PropTypes.number.isRequired,
  previewImage: PropTypes.string.isRequired,
  images: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isPremium: PropTypes.bool.isRequired,
  rating: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  maxAdults: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  goods: PropTypes.arrayOf(PropTypes.string),
  host: PropTypes.shape({
    id: PropTypes.number.isRequired,
    isPro: PropTypes.bool,
    name: PropTypes.stringisRequired,
    avatarUrl: PropTypes.string,
  }),
  description: PropTypes.string,
  location: PropTypes.shape({
    latitide: PropTypes.number,
    longitude: PropTypes.number,
    zoom: PropTypes.number,
  }),
};

export const reviewPropTypes = {
  id: PropTypes.number.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    isPro: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    avatarUrl: PropTypes.string.isRequired,
  }),
  rating: PropTypes.number.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

export const cityPropTypes = {
  name: PropTypes.string.isRequired,
  location: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    zoom: PropTypes.number.isRequired,
  }),
};

export const offerCardForTests = {
  city: {
    name: ``,
    location: {
      latitude: 1,
      longitude: 1,
      zoom: 1,
    },
  },
  id: 0,
  previewImage: ``,
  images: [``],
  title: ``,
  isFavorite: false,
  isPremium: false,
  rating: 0,
  type: ``,
  bedrooms: 0,
  maxAdults: 0,
  price: 0,
  goods: [``],
  host: {
    id: 0,
    isPro: false,
    name: ``,
    avatarUrl: ``,
  },
  description: ``,
  location: {
    latitude: 1,
    longitude: 1,
    zoom: 1,
  },
};

export const reviewForTests = {
  id: 0,
  user: {
    id: 0,
    isPro: false,
    name: ``,
    avatarUrl: ``,
  },
  rating: 0,
  comment: ``,
  date: `Thu Nov 07 2019 13:56:54 GMT+0300 (Москва, стандартное время)`,
};
