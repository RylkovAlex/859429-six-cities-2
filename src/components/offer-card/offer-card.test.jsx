import React from 'react';
import renderer from 'react-test-renderer';
import OfferCard from './offer-card';
import {MemoryRouter} from 'react-router';
import {offerCardForTests} from '../../prop-types/prop-types';

it(`PlaceCard correctly renders after relaunch`, () => {
  const props = {
    card: offerCardForTests,
    onCardHover: jest.fn(),
  };

  const tree = renderer
    .create(
        <MemoryRouter>
          <OfferCard {...props}/>
        </MemoryRouter>
    )
    .toJSON();


  expect(tree).toMatchSnapshot();
});
