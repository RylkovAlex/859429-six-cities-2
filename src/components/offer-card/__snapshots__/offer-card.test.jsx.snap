// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard correctly renders after relaunch 1`] = `
<article
  className="place-card cities__place-card"
  onMouseEnter={[Function]}
>
  
  <div
    className="place-card__image-wrapper cities__image-wrapper"
  >
    <a
      exact="true"
      href="/offer/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="/img/apartment-01.jpg"
        width={260}
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          20
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "56%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        exact="true"
        href="/offer/1"
        onClick={[Function]}
      >
        Beautiful & luxurious studio at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;
